### ISSUE TITLE
Requesting image size that does not exists yields no error

### ISSUE THREAD

**Issue by Arturszott** [FdezRomero/request-image-size#8]
Hey! Great library! Will save some KBs for our customers.

I've encountered one issue with requesting non existing image. As there is no `data` event `imageSizeError` will never be assigned. 

Correct me If i'm wrong :)

**Comment by FdezRomero** [comment]
Hi @Arturszott!

The `imageSizeError` variable stores the last error given by `imageSize(buffer)` if the response contains data, and then used if the image is completely downloaded but we were still unable to get the size.

In the case that you request a non-existing image, the `res` (response) object should emit an `error` event, which would reject the promise with the appropriate error ([check line 53](https://github.com/FdezRomero/request-image-size/blob/master/index.js#L53)).

Are you having any problems with non-existing images not rejecting the promise? If so, can you provide an example?

Thanks,
Rodrigo

**Comment by Arturszott** [comment]
Promise is rejected, but there is no error at all, it's undefined. This 'error' event is not emitted. I tried to log it inside the library and this particular place was not called.

**Comment by FdezRomero** [comment]
Can you provide the specific URL you're using? I tried with a non-existing URL and a `TypeError` is returned (in this case, the server responded with a 404 page in HTML, which is not supported):

```js
const requestImageSize = require("request-image-size");

requestImageSize('http://nodejs.org/images/doesntexist.png')
.then(size => console.log(size))
.catch(err => console.error(err));
```

And returns the error:

```
TypeError: unsupported file type: undefined (file: undefined)
```

Thanks!

**Comment by Arturszott** [comment]
I guess it's a matter of resource serving:

`https://media-cdn.holidaycheck.com/ugc/images/123` this gives undefined, while
`https://media-cdn.holidaycheck.com/ugc/images/f3b0c075-7e82-31bd-80e1-d4d45b41315b` is okay

I tried `http://nodejs.org/images/doesntexist.png` as well and I can confirm that it returns an error :)

**Comment by FdezRomero** [comment]
I think I found the cause. When you request `https://media-cdn.holidaycheck.com/ugc/images/123`, the server returns the following:

```
$ curl -v https://media-cdn.holidaycheck.com/ugc/images/123
*   Trying 23.60.217.198...
* TCP_NODELAY set
* Connected to media-cdn.holidaycheck.com (23.60.217.198) port 443 (#0)
* TLS 1.2 connection using TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
* Server certificate: san.cloudinary.com
* Server certificate: Let's Encrypt Authority X3
* Server certificate: DST Root CA X3
> GET /ugc/images/123 HTTP/1.1
> Host: media-cdn.holidaycheck.com
> User-Agent: curl/7.54.0
> Accept: */*
>
< HTTP/1.1 404 Not Found
< Access-Control-Allow-Headers: X-Requested-With
< Access-Control-Allow-Origin: *
< Access-Control-Expose-Headers: X-Cld-Error
< Content-Disposition: inline
< Content-Transfer-Encoding: binary
< Content-Type: image/gif
< Server: cloudinary
< Status: 404 Not Found
< X-Cld-Error: Resource not found - ugc/images/123
< X-Request-Id: 6d174a4c577eb588
< X-UA-Compatible: IE=Edge,chrome=1
< Content-Length: 0
< Expires: Thu, 21 Sep 2017 16:34:27 GMT
< Cache-Control: max-age=0, no-cache, no-store
< Pragma: no-cache
< Date: Thu, 21 Sep 2017 16:34:27 GMT
< Connection: keep-alive
<
* Connection #0 to host media-cdn.holidaycheck.com left intact
```

Notice that `Content-Length` is zero, meaning that no data is sent and `imageSize` never gets called, and therefore, no error is returned.

Maybe it should return a custom error for this specific case. Any thoughts?

**Comment by dustingraves** [comment]
I would like the custom errors for 404s etc.

**Comment by dustingraves** [comment]
My proposal as seen in this pull request: https://github.com/FdezRomero/request-image-size/pull/9

**Comment by FdezRomero** [comment]
@Arturszott @dustingraves I just published version 2.1.0 to npm with a fix that rejects the promise with an `HttpError` containing the status code and message if the response fails with an 4xx/5xx code.

Thanks for contributing!

**Comment by guillaumearm** [comment]
Hi 👋

Just for informations, I have the same error with this url: https://app-cdn.clickup.com/assets/favicons/apple-touch-icon.png

- node version: v12.9.1
- `request-image-size` version: v2.1.0

I just checked with `image-size` and it seems to not working too, I just created an issue on the image-size repository: https://github.com/image-size/image-size/issues/280

stay tuned 😉
